snuba/snuba_migrations/outcomes/0007_outcomes_add_indexes_and_codecs.py

```diff
@@ -0,0 +1,76 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import UUID, Column
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Adds indexes and codecs to match schema in SaaS. Clickhouse 20 doesn't
+    support adding codecs to key columns, so we don't add codecs to timestamp.
+    """
+
+    blocking = True
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.ModifyColumn(
+                StorageSetKey.OUTCOMES,
+                "outcomes_raw_local",
+                Column(
+                    "event_id",
+                    UUID(
+                        Modifiers(
+                            nullable=True,
+                            codecs=["LZ4HC(0)"],
+                            ttl="timestamp + toIntervalDay(30)",
+                        )
+                    ),
+                ),
+                target=operations.OperationTarget.LOCAL,
+            ),
```

Comment:
I would recommend you move out the modify column to its own migration. The migration is applying TTL to a column (which is a heavy operation) and also adding indexes (which is a lighter operation). Even though the end state would be the same whether you write 1 migration or 2, you separate out DDL changes by the type of operation being performed.

Reply:
moved to 0007_outcomes_add_event_id_ttl_codec